### PR TITLE
Add support for visionOS, macOS and tvOS

### PR DESCRIPTION
**Pull request by Supereg**
# Add support for visionOS, macOS and tvOS

## :recycle: Current situation & Problem
This PR adds support for visionOS, macOS and tvOS. These are not major changes and only required changes within the Package.swift file.

## :gear: Release Notes 
* Add support for visionOS, macOS and tvOS

## :books: Documentation
-

## :white_check_mark: Testing
We are currently testing iOS, watchOS and visionOS in CI.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PSchmiedmayer** [comment]
Thank you @Supereg! It would be great if we add some CI setup from https://github.com/StanfordBDHG/SwiftPackageTemplate to this repo to fully validate the functionality. We might need to add tvOS to https://github.com/StanfordBDHG/SwiftPackageTemplate before that.

In the same flow, it would be great if we can add all the badges to the README here that are also in https://github.com/StanfordBDHG/SwiftPackageTemplate.

**Comment by Supereg** [comment]
I just added tvOS and macOS builds for the Swift package. Also I added back CodeQL support with this PR 👍

**Comment by Supereg** [comment]
@PSchmiedmayer something I noticed. We specified Apple Watch Series 8 previously and as it doesn't exist in the toolchain anymore (now Series 9) it silently used a different platform (in this case macOS). We might want to see if we can move this into the action itself at some point.

File: CITATION.cff
Comment (PSchmiedmayer) [inline CITATION.cff:1]: Thank you @Supereg 🚀

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements @Supereg! 🚀

**Pull request merged**